pep-0704.rst

```diff
@@ -0,0 +1,137 @@
+PEP: 704
+Title: Require virtual environments by default for installers
+Author: Pradyun Gedam <pradyunsg@gmail.com>
+Sponsor: Brett Cannon <brett@python.org>
+PEP-Delegate: Paul Moore <p.f.moore@gmail.com>
+Discussions-To: https://discuss.python.org/t/22846
+Status: Draft
+Type: Standards Track
+Topic: Packaging
+Content-Type: text/x-rst
+Created: 16-Jan-2023
+Python-Version: 3.13
+Post-History: 16-Jan-2023
+
+
+Abstract
+========
+
+This PEP proposes that package installers like ``pip`` MUST require a virtual environment by default, on Python 3.13+.
+
+Motivation
+==========
+
+Python virtual environments are an essential part of the development workflow for Python. However, it requires extra effort since it is an opt-in feature, and requires users to either:
+
+- take explicit steps to activate/deactivate a virtual environment
+- use ``<path-to-venv>/<bin-path>/<executable>`` to run files
+
+This creates a barrier to entry for many new users, since virtual environments serve as a pre-requisite for typical Python development workflows and how virtual environments work is a lot of information for anyone new. It can take a lot of extra time and effort to explain them.
+
+Further, activation of virtual environment uses a slightly different syntax/mechanism to activate on different platforms.
+
+It also creates a scope for mistakes, since users must remember to activate the virtual environment before running an installer like ``pip``. On certain Linux distributions, this can result in the installer modifying files that are owned by the operating system (this is partially mitigated by :pep:`668` for distributions that opt-in to marking their environments accordingly).
+
+This PEP proposes that the default behaviour of installers like ``pip`` should be to require a virtual environment to be active. This would make it easier for new users to get started with Python, and would reduce the scope for mistakes.
+
+
+Specification
+=============
+
+When a user runs an installer without an active virtual environment, the installer MUST print an error message and exit with a non-zero exit code. The error message MUST inform the user that a virtual environment is required, SHOULD provide shell-specific instructions on how to create and activate a virtual environment named ``.venv``, and SHOULD provide a link to a documentation page that explains how to create and activate a virtual environment.
```

Comment:
What counts as a "virtual environment"? Only an environment created with `venv`/`virtualenv`? What about Conda environments? Or PEP 582 environments? Or other types of isolated environments? The PEP should either explicitly define this, or link (`:term:`) to a precise and authoritative definition, e.g. in the PyPA glossary.